kernel/arch/dreamcast/kernel/startup.s

```diff
@@ -180,8 +185,98 @@ normal_exit:
 	rts
 	nop
 
+#ifdef PROFILE
+! Call profiler startup code
+	.align	2
+_gprof_init:
+	mov.l monstartup_addr, r0
+	mov.l lowpc, r4
+	mov.l highpc, r5
+	jsr @r0
+	nop
+	! Register handler for trapa #33
+	mov.l _trapa_set_handler, r0
+	mov.l #33, r4
+	mov.l mcount_handler_addr, r5
+	mov.l #0, r6
+	jsr @r0
+	nop
+	rts
+	nop
+
+! 
+	.align	2
+_gprof_shutdown:
+	mov.l mcleanup_addr, r0
+	jsr	@r0
+	nop
+	rts
+	nop
+
+	.align	2
+mcount_handler:
+	! Save r4..r7
+	mov.l	r4, @-r15
+	mov.l	r5, @-r15
+	mov.l	r6, @-r15
+	mov.l	r7, @-r15
+	sts.l	pr, @-r15
+
+	sts pr, r4      ! Save previous program counter to r4 (frompc)
+	stc spc,r5	    ! Save current program counter to r5 (selfpc)
+
+	! The code sequence emitted by gcc for the profiling trap is
+	! .long 0   --- This is a 4-byte no-op placeholder to align trapa
+	! trapa #33 --- This is a 2-byte instruction
```

Comment:
If this is going to be reserving `TRAPA #33` specifically, maybe we should put a `#define` for the TRAPA index in the public header somewhere, so people will know it's reserved and don't step on it, and/or it can easily be configured later?

...unless this is some "no shit" standard TRAPA number everyone else knows about except me... In which case I'd still probably note it in the Doxygen.

Reply:
Yeah I can do that.  The `trapa #33` is reserved by the GCC compiler for sh.